### ISSUE TITLE
my $GOPATH is clobbered with a wrong dir after running :GoDef

### ISSUE THREAD

**Issue by andreimatei** [fatih/vim-go#1274]
### Behavior

I start vim, `:!echo $GOPATH` is `/Users/andrei/work `. I run `:GoDef` on any symbol and I'm taken to the definition. But then I run `:!echo $GOPATH` and I get something new and bogus: `/Users/andrei/work/src/github.com/cockroachdb/:/Users/andrei/work`. Now, with the wrong $GOPATH, much pain ensues.

Running with `let g:go_def_mode = 'godef'` makes this bad behavior go away.

I'm thinking maybe there's something wrong with the `$GOPATH` manipulation code around here:
https://github.com/fatih/vim-go/blob/c71c75e4c15b95759b6a8dd2f773edcb12ad0edc/autoload/go/def.vim#L5

### Steps to reproduce:

This doesn't repro for other folks around, but it happens reliably to me. I'm happy to work with you to see what's up.

### Configuration

`pwd` prints `/Users/andrei/work/src/github.com/cockroachdb/cockroach`. My `$GOPATH` is `/Users/andrei/work`. FWIW, our project is pretty complex and, under `.../cockroachdb/cockroach` (which, note, is one level down from what I get in the mangled $GOPATH), we have dirs like `pkg, vendor, bin`.

* `vimrc` you used to reproduce
   - I've reproduced with a minimal .vimrc having only vim-go in it

* vim version:
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Apr 19 2017 03:58:09) on OSX

* vim-go version:
commit b4936d89bd1480a864382108b49ebf5d2d16728a
Date:   Sun Apr 2 20:45:40 2017 -0400

* go version:

`go version go1.8 darwin/amd64`

**Comment by fatih** [comment]
Hi @andreimatei 

Can you try to add this:

```
let g:go_autodetect_gopath = 0
```

Also can you try to use this PR changes as well? https://github.com/fatih/vim-go/pull/1280

**Comment by andreimatei** [comment]
Hi @fatih

`let g:go_autodetect_gopath = 0` seems to fix my problem.

The problem is fixed with _and without_ #1280 (I've tested with what code I was running before, and with the new code after you've merged #1280). Is this surprising?

Without `let g:go_autodetect_gopath = 0`, the problem's still there even after #1280.

Is there still a bug to be investigated here?

Thanks!

**Comment by fatih** [comment]
I'm not sure how to fix it. I've double checked it and it should be still working. I'm closing this as I think it's solved with the above setting. However feel free to open a new issue if this is not fixed or you need to enable this setting. Thanks.
